src/field/arena.ts

```diff
@@ -189,30 +177,104 @@ export class Arena {
     return ret;
   }
 
+  /**
+   * Determines whether or not to reroll for the given {@linkcode PokemonSpecies} and level
+   * @param pokemonSpecies
+   * @param level
+   *
+   * Mega/Primal/Ultra legendaries and Arceus (720) cannot spawn below level 90
+   * Most legendaries (including Regigigas but not Kyurem, Zacian and Zamazenta) cannot spawn below level 70
+   * All other sublegends cannot spawn below level 50
+   * The final base case only has Cosmoem/Cosmog
+   * @returns `true` if rerolling is required, `false` otherwise
+   */
+  determineRerollIfLegendLike(pokemonSpecies: PokemonSpecies, level: number): boolean {
+    if (pokemonSpecies.isLegendLike()) {
+      if (pokemonSpecies.baseTotal >= 720) {
+        return level < 90;
+      } else if (pokemonSpecies.baseTotal >= 670) {
+        return level < 70;
+      } else if (pokemonSpecies.baseTotal >= 580) {
+        return level < 50;
+      } else {
+        return level < 30;
+      }
+    }
+    return false;
+  }
+
+  /**
+   * Generates a boss {@linkcode BiomePoolTier} for a given
+   * @param tierValue number from 0-63
+   *
+   * |         | tier values | Chance |
+   * |---------|-------------|--------|
+   * | Boss    | 20-63       | 44/64  |
+   * | Boss R  | 6-19        | 14/64  |
+   * | Boss SR | 1-5         | 5/64   |
+   * | Boss UR | 0           | 1/64   |
+   *
```

Comment:
```suggestion
   * Generates a boss {@linkcode BiomePoolTier} for a given tier value.
   * ```
   * |         | tier values | Chance |
   * |---------|-------------|--------|
   * | Boss    | 20-63       | 44/64  |
   * | Boss R  | 6-19        | 14/64  |
   * | Boss SR | 1-5         | 5/64   |
   * | Boss UR | 0           | 1/64   |
   * ```
   * @param tierValue - Number from 0-63
```